lib/usb/usb_private.h

```diff
@@ -157,6 +158,7 @@ struct _usbd_driver {
 				   void *buf, uint16_t len);
 	void (*poll)(usbd_device *usbd_dev);
 	void (*disconnect)(usbd_device *usbd_dev, bool disconnected);
+	void (* enable_sof_callback)(usbd_device *usbd_dev);
```

Comment:
Even with what you've got here, the backend method is _not_ "enable_callback", you're enabling the IRQ in the backend method.  The frontend method is enabling the callback (and providing said callback) the frontend is just asking to enable the SOF IRQ so that it can dispatch the callback.